### DISCUSSION TITLE
Reason of why some songs fail when trying to download?

### DISCUSSION THREAD

**Discussion by IsPherz** [Shabinder/SpotiFlyer#134] [Q&A]
I tried to download one of my Spotify playlists on the Android app a few of them get the X and after I try to download them again they still fail, theres any reason for that? mostly they are songs from japanese bands so its maybe a copyright issue? I will post screenshot in here 
![WhatsApp Image 2021-05-24 at 11 49 26 AM](https://user-images.githubusercontent.com/11274437/119380307-2a860d80-bc86-11eb-8e33-65ea112c41c4.jpeg)

**Comment by Shabinder** [comment]
The Reason is that the app couldnt find a matching song from youtube music. If you believe the song with X is available on youtube music too , Share both link , spotify's and youtube music's  and I can maybe improve youtube matching algo currently being used

**Reply by IsPherz** [reply]
oh I didnt know the app worked like that, so its not really downloading the music from spotify but matching the list and finding the music on YT?

**Reply by Shabinder** [reply]
Yeah for now , 

I have almost completed adding **jio saavn** as **another source** and hence we can get 320 Kbps music , if available and download songs which app couldn't find on yt.

**Reply by IsPherz** [reply]
at first I tought this literally converteed or extracted the spotify music, but after listening some songs in this list for example I realized they are different and not the version on spotify so still if they get found in youtube that doesnt ensure that is the actual song, in my case the songs that werent the spotify versions were like live audio from concerts or perfomances

**Reply by Shabinder** [reply]
Yeah , @FelipeCardona95 there is a room of error in the matching algo, 

Still there are many checks involved in order to prevent mismatching as much as possible , but still its not 100% robust.
As Jio-Saavn is ready and infuture I will be adding more sources and hence this room of error should be getting smaller and smaller with each new source.

So Stay Tuned.
